config/routes.rb

```diff
@@ -3,4 +3,20 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  # a route that maps the URL "/users/:author_id/posts"
+  # to the "index" action of the "posts" controller.
+   # It a Nested route to access posts for a specific user
+  get "/users/:author_id/posts", to: "posts#index", as: :user_posts
+ # a route that maps the URL "/posts/:id" to the "show"
+ # action of the "posts" controller.
+  get "/posts/:id", to: "posts#show", as: :user_post
+  # a route that maps the URL "/users" to the "index" action
+  # of the "users" controller. This means that when a user
+  # visits the "/users" URL, the "index" action in the
+  #"users" controller will be executed.
+  get "/users", to: "users#index"
+  # a route that maps the URL "/users/:id" to the "show"
+  # action of the "users" controller.
+  get "/users/:id", to: "users#show", as: :user
```

Comment:
- [OPTIONAL] You can also create the routes in this way which is shorter and make it looks better:

```Ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thank you very much for this suggestion @IngDavidTM